f8a_version_comparator/base.py

```diff
@@ -0,0 +1,35 @@
+#!/usr/bin/env python
+
+# Copyright © 2018 Red Hat Inc.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Item class acting as base class for various item types."""
+
+
+from abc import ABCMeta, abstractmethod
+
+
+class Item(metaclass=ABCMeta):
+    """Base class for maven version comparator tasks."""
+
+    @abstractmethod
+    def compare_to(self, item):
+        """Compare two maven versions."""
+        raise NotImplementedError()
```

Comment:
I think you can say just `pass` here as it shouldn't be possible to instantiate class with abstract methods anyway.

Reply:
Actually I'd suggest to make the function empty, ie. put the docstring in it, nothing more :)
Instead you won't be able to cover it by unit tests.

Reply:
Thanks @msrb and @tisnik for the reviews :). One quick question. These are abstract method definitions, If I just put a docstring and if this method is later on not implemented, it won't raise any error. Please correct me if I am wrong. :)

Reply:
@msrb @GeetikaBatra it does, see the [gist](https://gist.github.com/CermakM/5cc4b6fd5da3b71135ce09fdee17d9e6) ... the only problem is it raises TypeError .. Personaly, I usually do write `raise NotImplementedError()`, since it is more explicit.

Reply:
Thanks @CermakM. Looks like changing it to pass will also work fine. :)